src/main/java/org/broadinstitute/consent/http/service/DraftService.java

```diff
@@ -0,0 +1,173 @@
+      throw new NotFoundException(
+          String.format("Draft with UUID %s not found.", draftUUID.toString()));
+    }
+    if (!user.getUserId().equals(draft.getCreateUser().getUserId()) && !user.hasUserRole(
+        UserRoles.ADMIN)) {
+      throw new NotAuthorizedException("User not authorized to modify resource.");
+    }
+    return draft;
+  }
+
+  public void deleteDraftsByUser(User user) {
+    Set<DraftInterface> userDrafts = findDraftsForUser(user);
+    for (DraftInterface draft : userDrafts) {
+      deleteDraft(draft, user);
+    }
+  }
+
+  public Set<DraftSummary> findDraftSummariesForUser(User user) {
+    return draftDAO.findDraftSummariesByUserId(user.getUserId());
+  }
+
+  public Set<DraftInterface> findDraftsForUser(User user) {
+    return draftDAO.findDraftsByUserId(user.getUserId());
+  }
+
+  private DraftInterface findDraftByDraftUUID(
+      UUID draftSubmissionUUID) throws SQLException {
+    return draftDAO.findDraftById(draftSubmissionUUID);
+  }
+
+  public DraftInterface addAttachments(DraftInterface draft, User user,
+      Map<String, FormDataBodyPart> files) throws SQLException {
+    draftFileStorageService.storeDraftFiles(draft.getUUID(), user, files);
+    draftDAO.updateDraftByDraftUUID(draft.getUUID(),
+        new Date().toInstant(), user.getUserId());
+    return findDraftByDraftUUID(draft.getUUID());
+  }
+
+  public void deleteDraftAttachment(DraftInterface draft, User user, Integer fileId)
+      throws SQLException {
+    Optional<FileStorageObject> fileStorageObjectToDelete = draft.getStoredFiles().stream()
+        .filter(fileStorageObject -> fileStorageObject.getFileStorageObjectId().equals(fileId))
+        .findFirst();
+    if (fileStorageObjectToDelete.isPresent()) {
+      draftFileStorageService.deleteStoredFile(fileStorageObjectToDelete.get(), user);
+      draftDAO.updateDraftByDraftUUID(draft.getUUID(),
+          new Date().toInstant(), user.getUserId());
+    } else {
+      throw new NotFoundException(
+          String.format("Draft attachment is not found.  Draft: %s, Attachment: %d",
+              draft.getUUID(), fileId));
+    }
+  }
+
+  public void deleteDraft(DraftInterface draft, User user)
+      throws RuntimeException {
+    jdbi.useHandle(handle -> {
+      try {
+        handle.useTransaction(handler -> {
+          draftDAO.deleteDraftByUUIDList(List.of(draft.getUUID()));
+          draft.getStoredFiles().forEach(fileStorageObject -> {
+            try {
+              draftFileStorageService.deleteStoredFile(fileStorageObject, user);
+            } catch (SQLException e) {
+              throw new RuntimeException(e);
+            }
+          });
+        });
+      } catch (Exception e) {
+        handle.rollback();
+      }
+      handle.commit();
+    });
+  }
+
+  public StreamingOutput draftAsJson(DraftInterface draft) {
+    Gson gson = GsonUtil.buildGson();
+    return output -> {
+      output.write("{ \"document\":".getBytes());
```

Comment:
Can this be constructed using multiline string interpolation? This would make it more readable.

Reply:
Does this work for you?
    String outputString =
        String.format(
        """
        { "document": %s
        , "meta": %s
        }
        """, draft.getJson(), gson.toJson(draft));
    return output -> {
        output.write(outputString.getBytes());

Reply:
(Deleted my last code.  I added tests to validate and it turns out that the quotes in the JSON are stripped with String.format()).  Going to leave as-is for now.